### DISCUSSION TITLE
Testing wizards in applications

### DISCUSSION THREAD

**Discussion by ju5t** [spatie/laravel-livewire-wizard#28] [Ideas]
Testing wizards in an application is not as easy as it looks. Our use case is pretty simple: we load initial state into the wizard and pass state onwards to each step; so they can be tested individually.

This package has some internal fun extensions that can help to test our wizards. I think at a minimum we should register these two macro's in the package's service provider:

- emitEvents: this is such a cool macro!
- <s>and something that's missing: retrieving state for a particular step from the wizard (the current `jsonContent` fetches state from HTML, which is awesome, but not useful in a live app 😄 )</s> you can do this with `get()` already.

If we have access to both macro's in our application we could test the entire flow. I can look at both; hopefully this week.

What do you guys think?

**Comment by freekmurze** [comment]
I'm welcome any things that would make wizards more testable 👍

**Comment by ju5t** [comment]
For future readers, see #29.
